### PR TITLE
add a cleanup loop for failed vm using configmap

### PR DESCRIPTION
**Pull request by SchSeba**
Before this PR we have an issue if the vm creation fails because there
is an validation error from the virt-api we didn't clean the allocation.

This PR introduce a cleanup loop into the system.
When we hit the create vm mutating webhook we create the regular
allocation but we also mark the mac address in  a waiting configmap.

We have a waiting look the will check if the object is removed from the
map and if is not after 30 second we assume the object wasn't saved
into the etcd (no controller event) so we release the object.

If we get an event from the controller then we remove the mac address
from the configmap.

This PR are also add/remove some empty lines in the code, also fix the import positions and fix the output when a functional test fails

### CODE REVIEW

**Comment by SchSeba** [comment]
@alonSadan please take a look

File: pkg/manager/manager.go [hunk 1]
```diff
@@ -49,18 +50,20 @@ type KubeMacPoolManager struct {
 	stopSignalChannel        chan os.Signal
 	podNamespace             string
 	podName                  string
+	waitingTime              int
```
Comment (phoracek) [inline pkg/manager/manager.go]: Please add a docstring explaining why we have this attribute

File: pkg/pool-manager/pool.go [hunk 2]
```diff
@@ -31,6 +31,7 @@ const (
 	RuntimeObjectFinalizerName = "k8s.v1.cni.cncf.io/kubeMacPool"
 	networksAnnotation         = "k8s.v1.cni.cncf.io/networks"
 	networksStatusAnnotation   = "k8s.v1.cni.cncf.io/networks-status"
+	vmWaitConfigMapName        = "kubemacpool-vm-configmap"
```
Comment (phoracek) [inline pkg/pool-manager/pool.go]: what is this CM good for?

File: pkg/names/names.go [hunk 3]
```diff
@@ -0,0 +1,15 @@
+package names
```
Comment (phoracek) [inline pkg/names/names.go]: could you introduce this change in a different commit please?

File: pkg/pool-manager/pool_test.go [hunk 4]
```diff
@@ -100,13 +103,7 @@ var _ = Describe("Pool", func() {
...
 	Describe("Pool Manager General Functions ", func() {
 		It("should create a pool manager", func() {
-			fakeClient := fake.NewSimpleClientset()
-			startPoolRangeEnv, err := net.ParseMAC("02:00:00:00:00:00")
-			Expect(err).ToNot(HaveOccurred())
-			endPoolRangeEnv, err := net.ParseMAC("02:FF:FF:FF:FF:FF")
-			Expect(err).ToNot(HaveOccurred())
-			_, err = NewPoolManager(fakeClient, startPoolRangeEnv, endPoolRangeEnv, false)
-			Expect(err).ToNot(HaveOccurred())
+			createPoolManager("02:00:00:00:00:00", "02:FF:FF:FF:FF:FF")
```
Comment (phoracek) [inline pkg/pool-manager/pool_test.go]: could you please do this change in a separate commit?

File: pkg/pool-manager/virtualmachine_pool.go [hunk 5]
```diff
@@ -333,6 +355,121 @@ func (p *PoolManager) revertAllocationOnVm(vmName string, allocations map[string
 	p.releaseMacAddressesFromInterfaceMap(allocations)
 }
 
+func (p *PoolManager) getOrCreateVmMacWaitMap() (map[string]string, error) {
```
Comment (phoracek) [inline pkg/pool-manager/virtualmachine_pool.go]: do you think it would make sense to keep these helpers in a separate module. this logic is not trivial and it would be easier to document there

File: pkg/manager/manager.go (hunk 1 above)
Comment (SchSeba) [reply pkg/manager/manager.go]: done

File: pkg/pool-manager/pool.go (hunk 2 above)
Comment (SchSeba) [reply pkg/pool-manager/pool.go]: This CM saved the allocated mac addresses in waiting state.

We need to use it in case of a race condition that our manager go down and It have that types of allocations so the new leader can continue from the same state.

File: pkg/pool-manager/virtualmachine_pool.go (hunk 5 above)
Comment (SchSeba) [reply pkg/pool-manager/virtualmachine_pool.go]: I will prefer to document it here as we are going to have the same functions for pods in the future.

I will add comments onto of every function

File: pkg/names/names.go (hunk 3 above)
Comment (SchSeba) [reply pkg/names/names.go]: done

File: pkg/pool-manager/pool_test.go (hunk 4 above)
Comment (SchSeba) [reply pkg/pool-manager/pool_test.go]: done

File: pkg/pool-manager/pool.go (hunk 2 above)
Comment (phoracek) [reply pkg/pool-manager/pool.go]: Sounds dangerous synchronisation-wise, but that's not different from the rest of kubemacpool... :D

File: pkg/pool-manager/virtualmachine_pool.go [hunk 6]
```diff
@@ -333,6 +355,124 @@ func (p *PoolManager) revertAllocationOnVm(vmName string, allocations map[string
...
+// so we release the virtual machine
+func (p *PoolManager) vmWaitingCleanupLook(waitTime int) {
+	c := time.Tick(3 * time.Second)
+	log.Info("starting cleanup loop for waiting mac addresses")
+	for _ = range c {
+		p.poolMutex.Lock()
+
+		configMap, err := p.kubeClient.CoreV1().ConfigMaps(names.MANAGER_NAMESPACE).Get(vmWaitConfigMapName, metav1.GetOptions{})
+		if err != nil {
+			// TODO: should we exit here?
```
Comment (phoracek) [inline pkg/pool-manager/virtualmachine_pool.go]: Do we need to resolve this?

Comment (SchSeba) [reply pkg/pool-manager/virtualmachine_pool.go]: we can remove this.

The next loop will try again

**Comment by SchSeba** [comment]
Open an issue to track the implementation https://github.com/K8sNetworkPlumbingWG/kubemacpool/issues/62

**Comment by alonSadan** [comment]
looks good!
